teuthology/suite/run.py

```diff
@@ -196,15 +196,59 @@ def choose_ceph_version(self, ceph_hash):
             log.info('skipping ceph package verification')
 
     def choose_teuthology_branch(self):
+        """Select teuthology branch, check if it is present in repo and
+        return the branch name value.
+
+        The branch name value is determined in the following order:
+
+        Use ``--teuthology-branch`` argument value if supplied.
+
+        Use ``TEUTH_BRANCH`` environment variable value if declared.
+
+        If yaml file ``qa/.teuthology`` can be found in suite repo
+        supplied with ``--suite-repo`` or ``--suite-dir`` and
+        has ``teuthology_branch`` record then use its value.
+
+        Use ``teuthology_branch`` value if it is set in the one
+        of the teuthology config files ``$HOME/teuthology.yaml``
+        or ``/etc/teuthology.yaml`` correspondingly.
+
+        Use ``--suite-branch`` argument value if supplied.
+
+        Use ``--ceph-branch`` argument value if supplied.
+
+        Use ``master`` if branch determined with ``--suite-branch``
+        or ``--ceph-branch`` are missing in the teuthology repo.
+
+        Generate exception if the determine branch is not present
```

Comment:
"determined" ?

Reply:
@badone is _supplied with_ better?

Reply:
I've used this definition:
https://www.macmillandictionary.com/dictionary/british/determine
_3. INTRANSITIVE/TRANSITIVE to calculate something, or to discover it by examining evidence_

Reply:
That's fine, but it should be "determined branch", not "determine branch".